### ISSUE TITLE
Incorrect file reference of @types

### ISSUE THREAD

**Issue by falsandtru** [Microsoft/TypeScript#9404]
**TypeScript Version:**  master

**Code**

On the TypeScript repo's root directory.

```ts
// index.ts
```

**Expected behavior:**

```
$ node built/local/tsc.js index.ts
```

**Actual behavior:** 

```
$ node built/local/tsc.js index.ts
node_modules/@types/del/index.d.ts(8,40): error TS2304: Cannot find name 'Promise'.
node_modules/@types/del/index.d.ts(9,62): error TS2304: Cannot find name 'Promise'.
node_modules/@types/del/index.d.ts(11,43): error TS2304: Cannot find name 'Promise'.
node_modules/@types/del/index.d.ts(12,65): error TS2304: Cannot find name 'Promise'.
```

**Comment by weswigham** [comment]
We don't have a straight promise implementation in our `node_modules/@types` directory (because we use the `@types/q` we include due to other includes to provide those types), which is why it errors when used outside out build context (as the global patch `type Promise<T> = Q.Promise<T>` is in our Gulpfile).

Workaround: This is just caused by out new automatic inclusion of `@types` directories in compilations (plus the fact that we now use `@types` to help type our Gulpfile build, so actually have an `@types` directory). Pass one of the `--typeRoots` or `--types` options to turn it off and not pull in the types (which is what we do in our own Jakefile now).

**Comment by falsandtru** [comment]
I tried `--types` option but it seems not to work, and `--typeRoots` option doesn't exist in the option list.

```
$ node built/local/tsc.js --types index.ts
Version 2.0.0
Syntax:   tsc [options] [file ...]

Examples: tsc hello.ts
          tsc --out file.js file.ts
          tsc @args.txt
```

**Comment by weswigham** [comment]
Hm, [type roots should exist](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/commandLineParser.ts#L340), (as does `--types`). They're recent additions which only [just got added to schemastore](https://github.com/Microsoft/TypeScript/issues/9355).

By the way, the issue with `node built/local/tsc.js --types index.ts` is that `--types` expects a list (so it looks like you're massing index as its only element). At present the [only way](https://github.com/Microsoft/TypeScript/issues/9409#issuecomment-229260851) to pass an empty list is by following `--types` with another command line option.

**Comment by falsandtru** [comment]
Thanks! I'm waiting for the fix of #9409.

Here is full list of options. `typeRoots` option doesn't exist.

```
$ node built/local/tsc.js -h
Version 2.0.0
Syntax:   tsc [options] [file ...]

Examples: tsc hello.ts
          tsc --out file.js file.ts
          tsc @args.txt

Options:
 --allowJs                           Allow javascript files to be compiled.
 --allowSyntheticDefaultImports      Allow default imports from modules with no
default export. This does not affect code emit, just typechecking.
 --allowUnreachableCode              Do not report errors on unreachable code.
 --allowUnusedLabels                 Do not report errors on unused labels.
 --baseUrl                           Base directory to resolve non-absolute modu
le names.
 -d, --declaration                   Generates corresponding '.d.ts' file.
 --experimentalDecorators            Enables experimental support for ES7 decora
tors.
 --forceConsistentCasingInFileNames  Disallow inconsistently-cased references to
 the same file.
 -h, --help                          Print this message.
 --init                              Initializes a TypeScript project and create
s a tsconfig.json file.
 --jsx KIND                          Specify JSX code generation: 'preserve' or
'react'
 --lib                               Specify library files to be included in the
 compilation:
                                       'es5' 'es6' 'es2015' 'es7' 'es2016' 'es20
17' 'dom' 'webworker' 'scripthost' 'es2015.core' 'es2015.collection' 'es2015.gen
erator' 'es2015.iterable' 'es2015.promise' 'es2015.proxy' 'es2015.reflect' 'es20
15.symbol' 'es2015.symbol.wellknown' 'es2016.array.include' 'es2017.object' 'es2
017.sharedmemory'
 --mapRoot LOCATION                  Specify the location where debugger should
locate map files instead of generated locations.
 --maxNodeModuleJsDepth              The maximum dependency depth to search unde
r node_modules and load JavaScript files
 -m KIND, --module KIND              Specify module code generation: 'commonjs',
 'amd', 'system', 'umd' or 'es2015'
 --moduleResolution                  Specify module resolution strategy: 'node'
(Node.js) or 'classic' (TypeScript pre-1.6).
 --newLine NEWLINE                   Specify the end of line sequence to be used
 when emitting files: 'CRLF' (dos) or 'LF' (unix).
 --noEmit                            Do not emit outputs.
 --noEmitOnError                     Do not emit outputs if any errors were repo
rted.
 --noFallthroughCasesInSwitch        Report errors for fallthrough cases in swit
ch statement.
 --noImplicitAny                     Raise error on expressions and declarations
 with an implied 'any' type.
 --noImplicitReturns                 Report error when not all code paths in fun
ction return a value.
 --noImplicitThis                    Raise error on 'this' expressions with an i
mplied 'any' type.
 --noImplicitUseStrict               Do not emit 'use strict' directives in modu
le output.
 --noUnusedLocals                    Report Errors on Unused Locals.
 --noUnusedParameters                Report Errors on Unused Parameters.
 --outDir DIRECTORY                  Redirect output structure to the directory.

 --outFile FILE                      Concatenate and emit output to single file.

 --preserveConstEnums                Do not erase const enum declarations in gen
erated code.
 --pretty                            Stylize errors and messages using color and
 context. (experimental)
 -p DIRECTORY, --project DIRECTORY   Compile the project in the given directory.

 --reactNamespace                    Specify the object invoked for createElemen
t and __spread when targeting 'react' JSX emit
 --removeComments                    Do not emit comments to output.
 --rootDir LOCATION                  Specify the root directory of input files.
Use to control the output directory structure with --outDir.
 --skipLibCheck                      Skip type checking of declaration files.
 --sourceMap                         Generates corresponding '.map' file.
 --sourceRoot LOCATION               Specify the location where debugger should
locate TypeScript files instead of source locations.
 --strictNullChecks                  Enable strict null checks.
 --suppressImplicitAnyIndexErrors    Suppress noImplicitAny errors for indexing
objects lacking index signatures.
 -t VERSION, --target VERSION        Specify ECMAScript target version: 'ES3' (d
efault), 'ES5', or 'ES2015'
 --traceResolution                   Enable tracing of the name resolution proce
ss.
 --types                             Type declaration files to be included in co
mpilation.
 -v, --version                       Print the compiler's version.
 -w, --watch                         Watch input files.
 @<file>                             Insert command line options and files from
a file.

```
